themes/pzq/pzq.theme.sh

```diff
@@ -0,0 +1,170 @@
+
+# Theme inspired on:
+#  - ys theme - https://github.com/ohmyzsh/ohmyzsh/blob/master/themes/ys.zsh-theme
+#  - rana theme - https://github.com/ohmybash/oh-my-bash/blob/master/themes/rana/rana.theme.sh
+#
+# by Ziqiang Pei<ziqiangpei@foxmail.com>
+
+# ----------------------------------------------------------------- COLOR CONF
+D_DEFAULT_COLOR="${white}"
+D_INTERMEDIATE_COLOR="${bold_white}"
+D_NUMBER_COLOR="${bold_blue}"
+D_USER_COLOR="${cyan}"
+D_SUPERUSER_COLOR="${red}"
+D_MACHINE_COLOR="${green}"
+D_DIR_COLOR="${bold_yellow}"
+D_GIT_DEFAULT_COLOR="${echo_white}"
+D_GIT_BRANCH_COLOR="${echo_cyan}"
+D_GIT_PROMPT_COLOR="${echo_red}"
+D_SCM_COLOR="${bold_yellow}"
+D_BRANCH_COLOR="${cyan}"
+D_CHANGES_COLOR="${white}"
+D_TIME_COLOR="${white}"
+D_DOLLOR_COLOR="${bold_red}"
+D_CMDFAIL_COLOR="${red}"
+D_VIMSHELL_COLOR="${cyan}"
+
+# ------------------------------------------------------------------ FUNCTIONS
+case $TERM in
+  xterm*)
+      TITLEBAR="\033]0;\w\007"
+      ;;
+  *)
+      TITLEBAR=""
+      ;;
+esac
+
+is_vim_shell() {
+  if [ ! -z "$VIMRUNTIME" ];
+  then
+    echo "${D_INTERMEDIATE_COLOR}on ${D_VIMSHELL_COLOR}\
+vim shell${D_DEFAULT_COLOR} "
+  fi
+}
+
+mitsuhikos_lastcommandfailed() {
+  code=$?
+  if [ $code != 0 ];
+  then
+    echo " ${D_DEFAULT_COLOR}C:${D_CMDFAIL_COLOR}\
+$code ${D_DEFAULT_COLOR}"
+  fi
+}
+
+# vcprompt for scm instead of oh-my-bash default
+demula_vcprompt() {
+  if [ ! -z "$VCPROMPT_EXECUTABLE" ];
+  then
+    local D_VCPROMPT_FORMAT="on ${D_SCM_COLOR}%s${D_INTERMEDIATE_COLOR}:\
+${D_BRANCH_COLOR}%b %r ${D_CHANGES_COLOR}%m%u ${D_DEFAULT_COLOR}"
+    $VCPROMPT_EXECUTABLE -f "$D_VCPROMPT_FORMAT"
+  fi
+}
+
+# checks if the plugin is installed before calling battery_charge
+safe_battery_charge() {
+  if [ -e "${OSH}/plugins/battery/battery.plugin.sh" ];
+  then
+    battery_charge
+  fi
+}
+
+prompt_git() {
+	local branchName='';
+
+	# Check if the current directory is in a Git repository.
+	if [ $(git rev-parse --is-inside-work-tree &>/dev/null; echo "${?}") == '0' ]; then
+		# Get the short symbolic ref.
+		# If HEAD isn’t a symbolic ref, get the short SHA for the latest commit
+		# Otherwise, just give up.
+		branchName="$(git symbolic-ref --quiet --short HEAD 2> /dev/null || \
+			git rev-parse --short HEAD 2> /dev/null || \
+			echo '(unknown)')";
+
+		echo -e "${D_GIT_DEFAULT_COLOR}on ${D_GIT_BRANCH_COLOR}${branchName} ";
+	else
+		return;
+	fi;
+}
+
+limited_pwd() {
+  # Max length of PWD to display
+  MAX_PWD_LENGTH=20
+
+  # Replace $HOME with ~ if possible 
+  RELATIVE_PWD=${PWD/#$HOME/\~}
+
+  local offset=$((${#RELATIVE_PWD}-$MAX_PWD_LENGTH))
+
+  if [ $offset -gt "0" ]
+  then
+      local truncated_symbol="..."
+      TRUNCATED_PWD=${RELATIVE_PWD:$offset:$MAX_PWD_LENGTH}
+      echo -e "${truncated_symbol}/${TRUNCATED_PWD#*/}"
+  else
+      echo -e "${RELATIVE_PWD}"
+  fi
+}
+
+prompt_python() {
+  # for virtualenv
+  if [[ $VIRTUAL_ENV ]]; then
+    virtualenv=`basename "$VIRTUAL_ENV"`
```

Comment:
```suggestion
    local virtualenv=$(basename "$VIRTUAL_ENV")
```

Prefer `$(...)` to ``` `...` ``` which is deprecated by POSIX.